### PR TITLE
retrospy: fix icon resolution

### PR DESCRIPTION
**Pull request by Naxdy**
## Description of changes

`placeholder "out"` produced a path to the derivation for the individual desktop icons, not to the root derivation. This PR fixes icon resolution by using a separate derivation for the icons.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Naxdy** [comment]
Result of `nixpkgs-review pr 311127` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>retrospy</li>
  </ul>
</details>

File: pkgs/by-name/re/retrospy/package.nix [hunk 1]
```diff
@@ -59,7 +67,6 @@ buildDotnetModule {
 
   postInstall = ''
     mkdir -p $out/share/retrospy
```
[the remark this answers is not in the document]
Comment (Naxdy) [reply pkgs/by-name/re/retrospy/package.nix]: good point, it is not!

**Review by nicoonoclaste — APPROVED** [review]
Looks good, provided everything still builds (waiting on `nixpkgs-review`)

File: pkgs/by-name/re/retrospy/package.nix [hunk 2]
```diff
@@ -57,18 +65,13 @@ buildDotnetModule {
 
   inherit executables;
 
-  postInstall = ''
-    mkdir -p $out/share/retrospy
-    ${builtins.concatStringsSep "\n" (map (e: "cp ./${e}.ico $out/share/retrospy/${e}.ico") executables)}
-  '';
```
Comment (jopejoe1) [inline pkgs/by-name/re/retrospy/package.nix:63]: Use something like convert from imagemagick to convert to png and install at `$out/share/icons/hicolor/"$size"x"$size"/apps/$e.png`

**Review by jopejoe1 — CHANGES_REQUESTED** [review]
A few things to correctly support icons on linux.
1. Icons should not use a path as this breaks theming
2. Icons should be installed to the hicolor icon theme
3. I don't think ico files are supported by desktop files

**Comment by Naxdy** [comment]
> A few things to correctly support icons on linux.
> 
>     1. Icons should not use a path as this breaks theming
> 
>     2. Icons should be installed to the hicolor icon theme
> 
>     3. I don't think ico files are supported by desktop files

Regarding (3), ico files are supported, at least on KDE. (1) and (2) make sense, I'll have a look at that for a separate PR.

**Comment by jopejoe1** [comment]
I listed the 3 one because only png, svg and xpm are supported according to spec, but some desktops like KDE are probaly supporting more formats than that are listed in the spec.

For ref see: https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html

**Pull request merged**